src/API/Trace/SpanContext.php

```diff
@@ -109,22 +26,22 @@ private function __construct(
         string $spanId,
         int $traceFlags,
         bool $isRemote,
-        API\TraceStateInterface $traceState = null
+        TraceStateInterface $traceState = null
     ) {
         // TraceId must be exactly 16 bytes (32 chars) and at least one non-zero byte
         // SpanId must be exactly 8 bytes (16 chars) and at least one non-zero byte
-        if (!self::isValidTraceId($traceId) || !self::isValidSpanId($spanId)) {
-            $traceId = self::INVALID_TRACE;
-            $spanId = self::INVALID_SPAN;
+        if (!SpanContextValidator::isValidTraceId($traceId) || !SpanContextValidator::isValidSpanId($spanId)) {
+            $traceId = SpanContextValidator::INVALID_TRACE;
+            $spanId = SpanContextValidator::INVALID_SPAN;
         }
 
         $this->traceId = $traceId;
         $this->spanId = $spanId;
         $this->traceState = $traceState;
         $this->isRemote = $isRemote;
-        $this->isSampled = ($traceFlags & self::SAMPLED_FLAG) === self::SAMPLED_FLAG;
+        $this->isSampled = ($traceFlags & self::TRACE_FLAG_SAMPLED) === self::TRACE_FLAG_SAMPLED;
         $this->traceFlags = $traceFlags;
-        $this->isValid = self::isValidTraceId($this->traceId) && self::isValidSpanId($this->spanId);
+        $this->isValid = SpanContextValidator::isValidTraceId($this->traceId) && SpanContextValidator::isValidSpanId($this->spanId);
```

Comment:
Can be moved into if-block above / should be moved into factory method to be able to reuse `::getInvalid()` singleton.